File: backend/src/main/java/pe/edu/upc/backend/Entities/CultivoFertilizante.java
package pe.edu.upc.backend.Entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "cultivo_fertilizante")
public class CultivoFertilizante {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cultivo_id", nullable = false)
    private Cultivo cultivo;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "fertilizante_id", nullable = false)
    private Fertilizante fertilizante;

    @Column(name = "fecha_aplicacion")
    private LocalDate fechaAplicacion;

    @Column(name = "cantidad")
    private Double cantidad;
}

File: frontend/src/app/models/cultivo-fertilizante.ts
import { Cultivo } from "./cultivo";
import { Fertilizante } from "./fertilizante";

export interface CultivoFertilizante {
    id: number;
    cultivo: Cultivo;           // objeto Cultivo completo
    fertilizante: Fertilizante; // objeto Fertilizante completo
    fechaAplicacion: string;    // LocalDate → string en el JSON
    cantidad: number;
}
